Block: Implement deprecated block.

StevenDufresne opened the pull request:
Related to #162 

This PR implements the deprecated code block.

## Screenshots
| Desktop | Mobile | 
| --- | --- |
| ![localhost_8888_reference_functions_absint_](https://user-images.githubusercontent.com/1657336/212662430-688199b6-b4c4-46a8-a3b9-604589b3df7d.png) |  ![localhost_8888_reference_functions_absint_(iPhone 12 Pro)](https://user-images.githubusercontent.com/1657336/212662423-af6fbabd-2dc0-4ce9-a656-a450b84a3859.png) |

## Testing
It's a bit tricky seeing that our local environment doesn't have deprecated functions. Here is how I tested.

1. Set [this](https://github.com/WordPress/wporg-developer/blob/c186671beab3622f3c11b632bd3e281963616b82/source/wp-content/themes/wporg-developer-2023/inc/template-tags.php#L1096) to false.
2. You should now see the default banner.
3. To check that html rendering will work as expected, select HTML from the live site. [Example](https://developer.wordpress.org/reference/functions/get_blog_list/). Paste it in the DOM.

**Replace the content with this:**
```
<div class="callout callout-warning"><p><span class="screen-reader-text">Warning:</span> This function has been deprecated. Use <a href="https://developer.wordpress.org/reference/functions/wp_get_sites"></a><a href="https://developer.wordpress.org/reference/functions/wp_get_sites/" rel="function">wp_get_sites()</a>  instead.</p>
</div>
```

StevenDufresne committed:
Implement deprecated block.

StevenDufresne committed:
Update tag to be a div.

outdoor2kode reviewed (approved):
👍 Tested and looks good.

ryelle reviewed (changes requested):
I don't see a design in the Developers mockup for this, but in the design library there are some callout boxes designed, so I imagine this should match those:

| Site | Mockup |
|----|------|
| <img width="705" alt="Screenshot 2023-01-16 at 6 03 16 PM" src="https://user-images.githubusercontent.com/541093/212777649-5b8baaab-58c4-455c-90dd-63fe65bd42d8.png"> | <img width="792" alt="Screenshot 2023-01-16 at 6 02 30 PM" src="https://user-images.githubusercontent.com/541093/212777683-39a726cf-f708-40b5-bd7f-0128c6a151f1.png"> |

I see this HTML is coming from `get_deprecated` which in turn is built from `WPorg_Handbook_Callout_Boxes->warning_shortcode`, but I think this is a good place to refactor `get_deprecated` to return if not block syntax, something using the block class names for colors etc.

StevenDufresne commented:
> I don't see a design in the Developers mockup for this, but in the design library there are some callout boxes designed, so I imagine this should match those:

Good catch. Can I get a link to the design library?

StevenDufresne commented:
I'm going to return later to refactor. There are higher-priority issues to work on at the moment. I'll leave this PR hanging until then.

StevenDufresne committed:
Return the content from get_deprecated only.

StevenDufresne committed:
Add new notice block markup.

StevenDufresne commented:
I updated the code to use the notice component currently in https://github.com/WordPress/wporg-mu-plugins/pull/322.

Relevant commits:
- [42d2d4e](https://github.com/WordPress/wporg-developer/pull/167/commits/42d2d4ef07f8b1c379cbf21995a307369232b75b)
- [35562d7](https://github.com/WordPress/wporg-developer/pull/167/commits/35562d77a5ddcd1342b6a4d6cf014193c8e16c2c)

### Screenshot
<img width="400" alt="Screen Shot 2023-01-19 at 3 40 50 PM" src="https://user-images.githubusercontent.com/1657336/213373786-e514a105-fbbb-4edf-bfb8-5eed68c84790.png">

StevenDufresne committed:
Fix css linter errors.

ryelle committed:
Header: Fix font size for breadcrumbs

ryelle committed:
Remove the unused CSS file

The `callout` style here no longer applies to the block, and once that's removed the file is empty and can be removed.

ryelle committed:
Update notice to a warning

ryelle reviewed (approved):
Refactor looks good — `get_deprecated` is used by `get_changelog_data`, but that passes in `$formatted = false` so it should be straightforward to update when the changelog block is built.

I've removed the now-unused CSS, which made the CSS file unnecessary. I also updated the notice type to warning to match the red style of the old notice.

<img width="732" alt="Screenshot 2023-01-20 at 12 36 08 PM" src="https://user-images.githubusercontent.com/541093/213766127-f9c6e8cc-5654-4ffa-8555-99d04e6f38a3.png">
